DualView/src/main/java/com/microsoft/device/display/samples/dualview/ui/view/RestaurantView.kt

```diff
@@ -0,0 +1,243 @@
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import com.microsoft.device.display.samples.dualview.R
+import com.microsoft.device.display.samples.dualview.models.Restaurant
+import com.microsoft.device.display.samples.dualview.models.restaurants
+import com.microsoft.device.display.samples.dualview.ui.theme.selectedBody1
+import com.microsoft.device.display.samples.dualview.ui.theme.selectedBody2
+import com.microsoft.device.display.samples.dualview.ui.theme.typography
+import com.microsoft.device.display.samples.dualview.utils.formatPriceRange
+import com.microsoft.device.display.samples.dualview.utils.formatRating
+import com.microsoft.device.dualscreen.twopanelayout.navigateToPane2
+
+const val outlinePadding = 25
+const val narrowWidth = 1100
+const val thumbnailWidth = 140
+val TextStyleKey = SemanticsPropertyKey<TextStyle>("TextStyleKey")
+var SemanticsPropertyReceiver.textStyle by TextStyleKey
+
+@Composable
+fun RestaurantViewWithTopBar(
+    isDualScreen: Boolean,
+    viewWidth: Int,
+    selectedIndex: Int,
+    updateSelectedIndex: (Int) -> Unit
+) {
+    Scaffold(
+        topBar = { RestaurantTopBar(isDualScreen) }
+    ) {
+        RestaurantView(viewWidth, selectedIndex, updateSelectedIndex)
+    }
+}
+
+@Composable
+fun RestaurantTopBar(isDualScreen: Boolean) {
+    TopAppBar(
+        modifier = Modifier.testTag(stringResource(R.string.restaurant_top_bar)),
+        actions = { if (!isDualScreen) RestaurantActionButton() },
+        title = {
+            Text(
+                text = stringResource(R.string.app_name),
+                style = TextStyle(
+                    fontSize = 19.sp,
+                    fontWeight = FontWeight.SemiBold,
+                    color = MaterialTheme.colors.onPrimary
+                )
+            )
+        }
+    )
+}
+
+@Composable
+private fun RestaurantActionButton() {
+    IconButton(onClick = { navigateToPane2() }) {
+        Icon(
+            painter = painterResource(R.drawable.ic_map),
+            contentDescription = stringResource(R.string.switch_to_map),
+            tint = MaterialTheme.colors.onPrimary
+        )
+    }
+}
+
+@Composable
+fun RestaurantView(viewWidth: Int, selectedIndex: Int, updateSelectedIndex: (Int) -> Unit) {
+    Column(
+        modifier = Modifier.padding(top = outlinePadding.dp, start = outlinePadding.dp, end = outlinePadding.dp),
+        verticalArrangement = Arrangement.spacedBy(15.dp)
+    ) {
+        Text(
+            text = stringResource(R.string.list_title),
+            style = typography.subtitle1
+        )
+        RestaurantListView(viewWidth, selectedIndex, updateSelectedIndex)
+    }
+}
+
+@Composable
+fun RestaurantListView(viewWidth: Int, selectedIndex: Int, updateSelectedIndex: (Int) -> Unit) {
+    // REVISIT: use a WindowSizeClass check instead?
+    val isSmallScreen = viewWidth < narrowWidth && viewWidth != 0
```

Comment:
@joyl1216 Also I was wondering what you thought we should do here? I was refactoring and noticed that viewWidth is being used, do you think we should change it to use the width `WindowSizeClass` instead?

Reply:
Good idea, I think we should.